quickwit/quickwit-common/src/tracker.rs

```diff
@@ -0,0 +1,108 @@
+use std::mem::MaybeUninit;
+use std::ops::Deref;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::{Arc, Mutex};
+
+use census::{Inventory, TrackedObject as InventoredObject};
+
+pub type TrackedObject<T> = InventoredObject<RecordUnacknoledgedDrop<T>>;
+
+#[derive(Clone)]
+pub struct Tracker<T> {
```

Comment:
this struct is indirectly tested, but ought to have proper UT. the rest should be okay